std/sumtype.d

```diff
@@ -678,7 +688,17 @@ public:
         {
             import core.lifetime : move;
 
-            rhs.match!((ref value) { this = move(value); });
+            rhs.match!((ref value) {
+                static if (isCopyable!(typeof(value)))
+                {
+                    // Workaround for https://issues.dlang.org/show_bug.cgi?id=21542
+                    this = __ctfe ? value : move(value);
```

Comment:
Wouldn't a better solution be to change the CTFE-version of move to make them executable at compile-time? What about replacing the memcpy with an array-copy of `void[]`?

Reply:
Doing a proper `move` in CTFE is impossible, because CTFE does not support reinterpreting casts (which includes casting to `void[]`). See PR #936 for related discussion. The best that can be done is to fall back to a copy, as is done here.

Reply:
Afaict, the core of this limitation is that the compiler is lacking the ability to automatically (internally) perform a move of `value` into `this` at

```d
this = value;
```

as this statment is the last statement that references value before this is returned. If the compiler had that ability that would unlock being able to use non-copyable types in CTFE. Ping, @andralex @RazvanN7 @WalterBright.

Reply:
Yes, if the compiler could automatically turn these last-use assignments into moves, the explicit calls to `move` and `forward` would no longer be necessary in the first place, so there would be no need to add workarounds to avoid calling them in CTFE.

Reply:
Indeed. That would be awesome.